fairmq/shmem/Socket.h

```diff
@@ -211,28 +249,21 @@ class Socket final : public fair::mq::Socket
         }
         int elapsed = 0;
 
-        // put it into zmq message
-        const unsigned int vecSize = msgVec.size();
-        ZMsg zmqMsg(vecSize * sizeof(MetaHeader));
-
-        // prepare the message with shm metas
-        MetaHeader* metas = static_cast<MetaHeader*>(zmqMsg.Data());
-
-        for (auto& msg : msgVec) {
+        for (auto const& msg : msgVec) {
             auto msgPtr = msg.get();
             if (!msgPtr) {
                 return static_cast<int>(TransferCode::error);
             }
             assertm(dynamic_cast<shmem::Message*>(msgPtr), "given mq::Message is a shmem::Message");   // NOLINT
-            auto shmMsg = static_cast<shmem::Message*>(msgPtr);   // NOLINT(cppcoreguidelines-pro-type-static-cast-downcast)
-            std::memcpy(metas++, &(shmMsg->fMeta), sizeof(MetaHeader));
         }
 
+        auto metaMsg = MakeMetaMsg(msgVec);
+
         while (true) {
             int64_t totalSize = 0;
-            int nbytes = zmq_msg_send(zmqMsg.Msg(), fSocket, flags);
+            int nbytes = zmq_send(fSocket, metaMsg.Data(), metaMsg.Size(), flags);
```

Comment:
@rbx Hm, this makes an additional copy, right? So, I should use a `ZMsg` here.

Reply:
True

Reply:
Switched all cases to `ZMsg` which obsoleted `UMsg` so I removed it again. Will need another round of testing, also need to cover the new code paths in some of our unit tests.